.github/workflows/daily-ci.yaml

```diff
@@ -19,22 +19,22 @@ name: Daily CI
 
 on:
   pull_request:
-    paths:
-      - ".github/workflows/daily-ci.yaml"
+    types: [opened, synchronize, reopened, labeled]
   schedule:
-    # Daily run this action at 1am
     - cron: '0 1 * * *'
   workflow_dispatch:
 
 jobs:
   build-docker-image:
     name: Build Docker Image
+    if: |
+      (github.event_name != 'pull_request') ||
+      (github.event.action == 'labeled' && github.event.label.name == 'needs-daily-ci') ||
+      (github.event.action != 'labeled' && contains(github.event.pull_request.labels.*.name, 'needs-daily-ci'))
```

Comment:
I think the `workflow_call` method in #694 maybe better since we can have several jobs in the daily ci workflow.

Reply:
We can actually run kvrocks.yaml job everyday.

Reply:
> every job need an `if` here

I don't get the point. It's not a regression.

Reply:
If we call `daily-ci.yaml` from`kvrocks.yaml`, not only the dependencies logic will mess, but also we should now handle `pull_request` event with `labeled` type in `kvrocks.yaml`, it will cause more complex logic.

In Pulsar and previous SkyWalking CI settings we do repeat some conditions as long as it won't grow too much. We can review this issue if it cause concrete trouble.